src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -283,6 +297,21 @@ public ISymbolNode HelperEntrypoint(HelperEntrypoint entrypoint)
             return symbol;
         }
 
+        private TypeDesc _systemICastableType;
+
+        public TypeDesc ICastableInterface
```

Comment:
NodeCache shouldn't carry around private implementation details of it's nodes that are likely to become orphaned should the implementation of the nodes change. It should only carry around nodes, because it's a cache of nodes. Placing this kind of stuff here makes it equally likely to become orphaned if we ever get rid of the concept of ICastable as placing it on CompilerTypeSystemContext, or a CoreLibBinder (if we ever have one). It's already pretty far away from the place where it is used. We might as well put it in a centralized location.

Cc @davidwrighton